bokehjs/src/lib/core/util/canvas.ts

```diff
@@ -57,7 +58,13 @@ export class CanvasLayer {
       }
     }
 
-    (this._ctx as any).layer = this
+    Object.assign(this._ctx, {
+      layer: this,
+      rect_bbox(this: CanvasRenderingContext2D, bbox: BBox): void {
+        const {x, y, width, height} = bbox
+        this.rect(x, y, width, height)
+      },
+    })
```

Comment:
seems a little weird to patch like this, is this expected to go away at some point?

Reply:
This is quite standard practice, as long as it doesn't conflict with existing APIs. What would be somewhat weird, it to override `rect()` with custom logic and API overrides. The alternative to this would be to create a new class and proxy existing methods and attributes manually, which would be a lot of code. It has to be done manually btw., because real proxies in JS are awfully slow.